### ISSUE TITLE
Simplify function signatures

### ISSUE THREAD

**Issue by maedoc** [maedoc/sddekit#131]
Some function signatures are too complex, and it's far too easy to forget order. If there's no point in creating an interface to support the function call, it could be as simple as converting
```c
void foo(int a, int b, int c, int d);
```
to
```c
struct foo_args { int a, b, c, d; };
void foo(struct foo_args *args);
```
where in C99 calling can be concise almost like keyword arguments,
```c
foo((struct foo_args) {.d=43, .c=2, .b=2, .a=2});
```
and in older dialects more verbose
```c
struct foo_args args;
args.a = 2;
args.b = 2;
args.c = 2;
args.d = 42;
foo(&args);
```
but we already assume C99 per variadic macros, so this isn't too much an issue.

**Comment by catchchaos** [comment]
This involves scanning the header files and changing all signatures. And then looking through the files and changing corresponding headers there too right?

**Comment by maedoc** [comment]
Yes, but I don't actually want to this yet. I'm still refactoring the API a little bit.
